templates/example_visualization.ipynb

```diff
@@ -0,0 +1,263 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "all_data = pd.read_csv(data_dir)\n",
+    "all_data.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Visualize Patient-Population Distribution Examples"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Visualzing PatientID vs Cell Type, using default paramters so showing all graphs\n",
+    "visualize.visualize_patient_population_distribution(all_data, \"PatientID\", \"cell_type\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Visualize SampleID vs cell_type, ommiting total distribution since population same as previous function\n",
+    "visualize.visualize_patient_population_distribution(all_data, \"SampleID\", \"cell_type\", show_total_count=False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Visualizing PatientID vs Cell Lineage, \n",
+    "visualize.visualize_patient_population_distribution(all_data, \"PatientID\", \"cell_lin\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Visualizing cell lineage vs cell type\n",
+    "visualize.visualize_patient_population_distribution(all_data, \"cell_lin\", \"cell_type\", show_total_count=False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Visualizing cell lineage vs cell type, saving it in the same directory\n",
+    "visualize.visualize_patient_population_distribution(all_data, \"cell_lin\", \"cell_type\", show_total_count=False, save_dir = \"\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# PCA Projection of data, with all the Target categories, and PatientID as the label. Saves image into current directory.\n",
+    "\n",
+    "dimensionality_reduction.visualize_dimensionality_reduction(all_data,['C', 'Na', 'Si', 'HH3',\n",
+    "       'Vimentin', 'SMA', 'Background', 'Collagen.1', 'Lag3', 'CD4', 'CD14',\n",
+    "       'Foxp3', 'PD.1', 'CD31', 'PD.L1', 'E.cadherin', 'Ki67', 'CD209',\n",
+    "       'CD206', 'gdTCR', 'iNOS', 'CD68', 'CD36', 'CD8', 'CD3', 'IDO', 'CD11c',\n",
+    "       'CD163', 'CD20', 'CD16', 'IFNg', 'HLA.DR.DQ.DP', 'CD11b', 'CD45',\n",
+    "       'H3K9Ac', 'Keratin.pan', 'CD103', 'MastChyTry', 'MPO', 'NaKATPase',\n",
+    "       'HLA.Class.1', 'Ta', 'Au'], \"PatientID\",algorithm=\"PCA\", save_dir=\"\")"
+   ]
```

Comment:
I think you should push a version of the notebook with a graph displayed for each of the `visualize_dimensionality_reduction` cells so we can see what a sample output would look like.